connector/connect/common/src/main/protobuf/spark/connect/relations.proto

```diff
@@ -905,6 +906,29 @@ message ApplyInPandasWithState {
   string timeout_conf = 7;
 }
 
+message FlatMapGroupsWithState {
```

Comment:
Add documentation

Reply:
Is it the same as `GroupMap`? Any reason that you cannot reuse the same message?

Reply:
If I want to reuse the existing `GroupMap`, do I just add new fields as optional? Also, what's the recommend way to add optional fields? I saw that in `GroupMap`, `sorting_expressions` is commented as optional, but it doesn't have an `optional` keyword in front.